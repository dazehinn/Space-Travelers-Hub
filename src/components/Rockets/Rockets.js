import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import styles from './Rockets.module.css';
import { fetchRockets } from '../../redux/rockets/rocketsSlice';

const Rockets = () => {
  const { rocketArray, isLoading } = useSelector((store) => store.rocket);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  if (isLoading) {
    return (
      <div className="loading">
        <h1>Loading...</h1>
      </div>
    );
  }

  return (
    rocketArray.map((rocket) => (
      <div key={rocket.id}>
        <div className={styles.rocketCard}>
          <img className={styles.rocketImg} src={rocket.flickr_images} alt="rocket" />
          <div className={styles.rocketInfo}>
            <h2>{rocket.name}</h2>
            <p>{rocket.description}</p>
            <button className={styles.reserveBtn} type="button">Reserve Rocket</button>
          </div>
        </div>
      </div>
    ))
  );
};

export default Rockets;
