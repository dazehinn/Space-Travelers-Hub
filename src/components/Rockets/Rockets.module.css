.rocketImg {
  width: 150px;
  padding-right: 12px;
}

.rocketCard {
  display: flex;
  padding: 0 25px 0 25px;
}

.rocketInfo {
  display: inline;
}
