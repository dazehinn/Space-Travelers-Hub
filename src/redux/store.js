import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import reducer from './missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: reducer,
    rocket: rocketReducer,
  },
});

export default store;
