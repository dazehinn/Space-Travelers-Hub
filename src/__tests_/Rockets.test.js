import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../components/Rockets/Rockets';
import store from '../redux/Store';

it('Test Rocket component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
