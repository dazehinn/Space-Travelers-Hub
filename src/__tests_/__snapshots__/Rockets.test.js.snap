// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rocket component renders correctly 1`] = `null`;
